packages/block-editor/src/components/block-list/use-block-props/index.js

```diff
@@ -87,8 +88,9 @@ export function useBlockProps( props = {}, { __unstableIsHtml } = {} ) {
 				isBlockMultiSelected( clientId ) ||
 				isAncestorMultiSelected( clientId );
 			const blockName = getBlockName( clientId );
+			const rootClientId = getBlockRootClientId( clientId );
 			return {
-				index: getBlockIndex( clientId ),
+				index: getBlockIndex( clientId, rootClientId ),
```

Comment:
It seems strange that the root client ID needs to be passed if it can be derived from the client ID.

Reply:
It might not be the only selector like that.